src/App.tsx

```diff
@@ -1,14 +1,43 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+function isIncludesTitle(title: string, query: string) {
+  return title.toLowerCase().includes(query.toLowerCase());
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState('All');
+  const [query, setQuery] = useState('');
+  const [todoId, setTodoId] = useState(0);
+
+  useEffect(() => {
+    getTodos()
+      .then(response => {
+        setTodos(response);
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterBy) {
+      case 'active':
+        return (todos.filter(todo => !todo.completed).filter(({ title }) => isIncludesTitle(title, query)));
```

Comment:
any way you filter two times )
```suggestion
        return (todos.filter(({ title, completed }) => !completed && isIncludesTitle(title, query)));
```